### ISSUE TITLE
Please clarify warning in fundamentals -> Injection scopes

### ISSUE THREAD

**Issue by chtpl** [nestjs/docs.nestjs.com#2586]
## I'm submitting a...

- [ ] Regression 
- [ ] Bug report
- [ ] Feature request
- [x] Documentation issue or request (new chapter/page)
- [ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.

## Current behavior
The documentation states 
```WARNING
Note this strategy is not ideal for applications operating with a large number of tenants.
```

## Expected behavior
The documentation should also add a sentence to that _why_ this is not considered ideal and what approach would be better.

**Comment by Tony133** [comment]
@chtpl would you like to create a PR for this? PRs are always welcome for core NestJS team.

A suggestion if you think the documentation needs to be updated or you would like to propose a change directly create a PR with your rationale, without creating an issue.

**Comment by chtpl** [comment]
If I knew a better advice on that specific topic, I could. Unfortunately my experience with NestJS is very limited - with no experience in a large scale use case

**Comment by smolinari** [comment]
@chtpl - Do you know about how dependency injection container (DIC) systems work? Because that is the knowledge that is assumed, with that warning. What happens is, with scoped injection, you no longer are working with singletons. And thus, when you need to re-instantiate for every request or across only a section of dependencies, you end up "bloating up" the DIC and slowing down Nest.  

That being said, the docs do go into the performance issues and also suggest the "Durable Provider" pattern to counteract the performance issues. However, the same performance issues you have with request scoped providers becomes an issue with durable providers too, once the numbers get too high i.e. too many tenants in the example. 

Basically the rule should be, don't do scoped injection, if you really don't have to. :) 

Scott

**Comment by Tony133** [comment]
Hi @chtpl,
Scott gave you a very nice, very summarized explanation, so I don't know what we can still improve in the documentation at the moment.  

Probably without some technical knowledge it is normal that it can become complicated at first to understand how some things work in the frameworks, but with a little study you will see that everything will be easier to understand :-)
